File: OPENAPI-REST-API/swagger-client/dart/lib/model/vps_extra_info_tables.dart
part of swagger.api;

class VpsExtraInfoTables {
  
  VpsIPInfo ipInfo = null;

  VpsExtraInfoTables();

  @override
  String toString() {
    return 'VpsExtraInfoTables[ipInfo=$ipInfo, ]';
  }

  VpsExtraInfoTables.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    ipInfo = new VpsIPInfo.fromJson(json['ip_info']);
  }

  Map<String, dynamic> toJson() {
    return {
      'ip_info': ipInfo
     };
  }

  static List<VpsExtraInfoTables> listFromJson(List<dynamic> json) {
    return json == null ? new List<VpsExtraInfoTables>() : json.map((value) => new VpsExtraInfoTables.fromJson(value)).toList();
  }

  static Map<String, VpsExtraInfoTables> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, VpsExtraInfoTables>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new VpsExtraInfoTables.fromJson(value));
    }
    return map;
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/VpsExtraInfoTables.cpp


#include "VpsExtraInfoTables.h"

using namespace Tiny;

VpsExtraInfoTables::VpsExtraInfoTables()
{
	ip_info = VpsIPInfo();
}

VpsExtraInfoTables::VpsExtraInfoTables(std::string jsonString)
{
	this->fromJson(jsonString);
}

VpsExtraInfoTables::~VpsExtraInfoTables()
{

}

void
VpsExtraInfoTables::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *ip_infoKey = "ip_info";

    if(object.has_key(ip_infoKey))
    {
        bourne::json value = object[ip_infoKey];




        VpsIPInfo* obj = &ip_info;
		obj->fromJson(value.dump());

    }


}

bourne::json
VpsExtraInfoTables::toJson()
{
    bourne::json object = bourne::json::object();






	object["ip_info"] = getIpInfo().toJson();


    return object;

}

VpsIPInfo
VpsExtraInfoTables::getIpInfo()
{
	return ip_info;
}

void
VpsExtraInfoTables::setIpInfo(VpsIPInfo ip_info)
{
	this->ip_info = ip_info;
}



